### ISSUE TITLE
Issues with how basic auth is being done when in Python 3

### ISSUE THREAD

**Issue by ashic** [puiterwijk/flask-oidc#49]
I'm seeing a problem anywhere this library is using client secret basic as the auth mechanism. The issue boils down to:

```
 bas = 'foo:bar'
 bytes = bytearray(bas, 'utf-8')
 s = 'Basic %s' % b64encode(bytes)
 s
"Basic b'Zm9vOmJhcg=='"
 s2 = 'Basic %s' % b64encode(bytes).decode('utf-8')
 s2
'Basic Zm9vOmJhcg=='
```
In python 3.6.4, serializing bytes is outputting b'actual_value', which is then failing authentication. Adding a .decode('utf-8') results in a string, and the b'' isn't added anymore. This doesn't manifest in python 2. 

The following works in both python 2 and 3:

```
 bas = 'foo:bar'
 bytes = bytearray(bas, 'utf-8')
 s = str('Basic %s' % b64encode(bytes).decode('utf-8'))
 s
```
Would this be something you're willing to incorporate? I can get a PR ready for this if so. (I think this affects https://github.com/puiterwijk/flask-oidc/issues/44 as well).

**Comment by marcvs** [comment]
I'm hitting the same bug here and need this to be fixed.

I'm also willing to generate PRs, if this project is still alive

What is the status of flask-oidc?

**Comment by puiterwijk** [comment]
Thanks to @marcvs's patch, this got fixed: 4bfd89149aadbdd98980720ae54d34a102fe7664
